go/obsclient/obsclient.go

```diff
@@ -0,0 +1,41 @@
+package obsclient
+
+import (
+	"math/big"
+
+	"github.com/ethereum/go-ethereum/common/hexutil"
+	"github.com/obscuronet/go-obscuro/go/rpcclientlib"
+)
+
+// ObsClient requires an RPC Client and provides access to general Obscuro functionality that doesn't require viewing keys.
+type ObsClient struct {
+	c rpcclientlib.Client
+}
+
+func Dial(rawurl string) (*ObsClient, error) {
+	rc, err := rpcclientlib.NewNetworkClient(rawurl)
+	if err != nil {
+		return nil, err
+	}
+	return NewClient(rc), nil
+}
+
+func NewClient(c rpcclientlib.Client) *ObsClient {
+	return &ObsClient{c}
+}
+
+func (oc *ObsClient) Close() {
+	oc.c.Stop()
+}
+
+// Blockchain Access
+
+// ChainID retrieves the current chain ID for transaction replay protection.
+func (oc *ObsClient) ChainID() (*big.Int, error) {
+	var result hexutil.Big
+	err := oc.c.Call(&result, "eth_chainId")
```

Comment:
There should be an existing constant for this. Can we use it? Then we avoid mismatches between the names used in tests, and the names used by the clients.